akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -61,7 +64,21 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
         props.args map serialize foreach builder.addArgs
-        props.args map (a ⇒ if (a == null) "null" else a.getClass.getName) foreach builder.addClasses
+        props.args.map { a ⇒
+          val argClassName =
+            if (a == null) "null"
+            else {
+              val className = a.getClass.getName
+              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && className.contains("$Lambda$")) {
+                // The serialization of the parameters is based on passing class name instead of
+                // serializerId and manifest as we usually do. With Scala 2.12 the functions are generated as
+                // lambdas and we can't use that load class from that name when deserializing.
```

Comment:
If it would have done it correctly I could have used `""` as the class manifest, but can't change due to compatibility.

Then the above `$Lamda$` hack would not be needed because it would pick the manifest from the serializer, which in that case would be no manifest (JavaSerializer includeManifest=false)

There is no nice way of seeing that it's a lamda, e.g. http://stackoverflow.com/questions/23870478/how-to-correctly-determine-that-an-object-is-a-lambda/23870800#23870800